projects/kit/components/input-date-time/input-date-time.component.ts

```diff
@@ -299,9 +294,10 @@ export class TuiInputDateTimeComponent
     }
 
     private updateNativeValue(day: TuiDay) {
-        const time = this.nativeValue.split(DATE_TIME_SEPARATOR)[1] || '';
+        const {timeMode, nativeValue} = this;
+        const time = nativeValue.split(DATE_TIME_SEPARATOR)[1] || '';
 
-        this.nativeValue = `${day.toString()}${DATE_TIME_SEPARATOR}${time}`;
+        this.nativeValue = this.getDateTimeString(day, time, {timeMode});
```

Comment:
If we need timeMode only for the TuiTime class, can we make the timeMode argument optional?

Reply:
done